platform/os_windows.go

```diff
@@ -257,43 +248,72 @@ func (p *execClient) ExecutePowershellCommandWithContext(ctx context.Context, co
 }
 
 // SetSdnRemoteArpMacAddress sets the regkey for SDNRemoteArpMacAddress needed for multitenancy if hns is enabled
-func SetSdnRemoteArpMacAddress(execClient ExecClient) error {
-	exists, err := execClient.ExecutePowershellCommand(CheckIfHNSStatePathExistsCommand)
+func SetSdnRemoteArpMacAddress() error {
+	// open the registry key
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, "SYSTEM\\CurrentControlSet\\Services\\hns\\State", registry.READ|registry.SET_VALUE)
 	if err != nil {
-		errMsg := fmt.Sprintf("Failed to check the existent of hns state path due to error %s", err.Error())
-		log.Printf(errMsg)
-		return errors.Errorf(errMsg)
+		if errors.Is(err, registry.ErrNotExist) {
+			return nil
+		}
+		return errors.Wrap(err, "could not open registry key")
 	}
-	if strings.EqualFold(exists, "false") {
-		log.Printf("hns state path does not exist, skip setting SdnRemoteArpMacAddress")
-		return nil
+	defer k.Close()
+	// set the key value
+	if err = setSDNRemoteARPMACAddress(k); err != nil {
+		return errors.Wrap(err, "could not set registry key")
 	}
-	if sdnRemoteArpMacAddressSet == false {
-		result, err := execClient.ExecutePowershellCommand(GetSdnRemoteArpMacAddressCommand)
-		if err != nil {
-			return err
-		}
-
-		// Set the reg key if not already set or has incorrect value
-		if result != SDNRemoteArpMacAddress {
-			if _, err = execClient.ExecutePowershellCommand(SetSdnRemoteArpMacAddressCommand); err != nil {
-				log.Printf("Failed to set SDNRemoteArpMacAddress due to error %s", err.Error())
-				return err
-			}
+	// connect to the service manager
+	m, err := mgr.Connect()
+	if err != nil {
+		return errors.Wrap(err, "could not connect to service manager")
+	}
+	defer m.Disconnect() //nolint:errcheck // ignore error
+	// open the HNS service
+	service, err := m.OpenService("hns")
+	if err != nil {
+		return errors.Wrap(err, "could not access service")
+	}
+	defer service.Close()
+	return errors.Wrap(restartService(service), "could not restart service")
```

Comment:
nit: While I understand that this will only actually return an error if `restartService` returns an error itself, it makes this harder to read at first unless you are familiar with `github.com/pkg/errors` (plus more complicated when you want to debug and step through code). I've been asking people to avoid this in networking aquarius for same readability issues. Good ol' 
```go
if err := restartService(service); err != nil {
    return errors.Wrap(err, "could not restart service")
}

return nil
```

reads clearly IMO.

Reply:
fixed